lib/Parser/parse.cpp

```diff
@@ -5423,6 +5455,17 @@ void Parser::ParseFncFormals(ParseNodePtr pnodeFnc, ushort flags)
                     {
                         Error(ERRRestWithDefault);
                     }
+
+                    // In defer parse mode we have to flag the function node to indicate that it has default arguments
+                    // so that it will be considered for any syntax error scenario.
+                    // Also mark it before parsing the expression as it may contain functions.
+                    ParseNode* currentFncNode = GetCurrentFunctionNode();
```

Comment:
Why not just m_currentFunctionNode? instead of getting this from GetCurrentFunctionNode()? this function will give different thing depending on which one is available. It may be possible that we may set this flag on the wrong functionNode. I know you just moved this code up - but good to verify that.